cli.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -e
+
+export PATH="$HOME/.cargo/bin:$PATH"
+
+# Jenkins will set WORKSPACE to the top level directory that contains
+# the stratis-cli git repo
+if [ -z "$WORKSPACE" ]
+then
+    echo "Required WORKSPACE environment variable not set"
+    exit 1
+fi
+
+export STRATIS_DEPS_DIR=$WORKSPACE/stratis-deps
+
+cd $WORKSPACE
+
+# If there is a stale STRATIS_DEPS_DIR remove it
+if [ -d $STRATIS_DEPS_DIR ]
+then
+    rm -rf $STRATIS_DEPS_DIR
+fi
+
+
+mkdir $STRATIS_DEPS_DIR
+cd $STRATIS_DEPS_DIR
+
+# Clone the dependencies
+git clone https://github.com/stratis-storage/dbus-python-client-gen.git
+git clone https://github.com/stratis-storage/dbus-client-gen.git
+git clone https://github.com/stratis-storage/into-dbus-python.git
+git clone https://github.com/stratis-storage/dbus-signature-pyparsing.git
+git clone https://github.com/stratis-storage/stratisd.git
+
+if [ ! -f  /etc/dbus-1/system.d/stratisd.conf ]
+then
+    cp $STRATIS_DEPS_DIR/stratisd/stratisd.conf /etc/dbus-1/system.d/.
+fi
+
+cd $STRATIS_DEPS_DIR/stratisd
+make build
+
+# Set the PYTHONPATH to use the dependencies
+cd $STRATIS_DEPS_DIR/dbus-client-gen
```

Comment:
You should check out the latest tagged version of every one of these dependencies. The same reasoning works for all. The reason the others are different is that they've been quiescent for more than a year in most cases and the only changes committed to master have been infrastructure maintenance. But if it ever was necessary to work on them this job would likely start to fail.

Reply:
I added a loop to check out the latest tagged version of dbus-client-gen, dbus-signature-pyparsing, dbus-python-client-gen, and into-dbus-python.